mcfly/modelgen.py

```diff
@@ -272,117 +343,338 @@ def generate_CNN_model(x_shape, class_number, filters, fc_hidden_nodes,
+
+    Parameters
+    ----------
+    input_shape : tuple
+        Shape of the input dataset: (num_samples, num_timesteps, num_channels)
+    class_number : int
+        Number of classes for classification task
+    min_filters_number : int
+        Number of filters for first convolutional layer
+    max_kernel_size: int,
+        Maximum kernel size for convolutions within Inception modul
+    network_depth : int
+        Depth of network, i.e. number of Inception modules to stac
+    learning_rate : float
+        learning rate
+    regularization_rate : float
+        regularization rate
+    metrics : list
+        Metrics to calculate on the validation set.
+        See https://keras.io/metrics/ for possible values.
+
+    Returns
+    -------
+    model : Keras model
+        The compiled Keras model
+    """
+
+    dim_length = input_shape[1]  # number of samples in a time series
+    dim_channels = input_shape[2]  # number of channels
+    weightinit = 'lecun_uniform'
+    regularization = 0
+
+    def conv_bn_relu_3_sandwich(x, filters, kernel_size):
+        first_x = x
+        for i in range(3):
+            x = layers.Convolution1D(filters, kernel_size, padding='same',
+                                     kernel_initializer=weightinit,
+                                     kernel_regularizer=l2(regularization))(x)
+            x = layers.BatchNormalization()(x)
+            x = layers.ReLU()(x)
+
+        #x = layers.Concatenate()([x, first_x])
+        first_x = layers.Convolution1D(filters, kernel_size=1, padding='same',
+                                     kernel_initializer=weightinit,
+                                     kernel_regularizer=l2(regularization))(x)
+        x = layers.Add()([x, first_x])
+        return x
+
+    x = layers.Input((dim_length, dim_channels))
+    inputs = x
+    
+    x = layers.BatchNormalization()(inputs) # Added batchnorm (not in original paper)
+
+    # Define/guess filter sizes and kernel sizes
+    # Logic here is that kernals become smaller while the number of filters increases
+    kernel_sizes = [max(3, int(max_kernel_size // (1.41 ** i))) for i in range(network_depth)]
+    filter_numbers = [int(min_filters_number * (1.41 ** i)) for i in range(network_depth)]
+
+    for i in range(network_depth):
+        x = conv_bn_relu_3_sandwich(x, filter_numbers[i], kernel_sizes[i])
+
+    x = layers.GlobalAvgPool1D()(x)
+    output_layer = layers.Dense(class_number, activation='softmax')(x)
+
+    # Create model and compile
+    model = Model(inputs=inputs, outputs=output_layer)
+
+    model.compile(loss='categorical_crossentropy',
+              optimizer=Adam(lr=learning_rate),
+              metrics=metrics)
+
+    return model
+
+
+def generate_InceptionTime_model(input_shape,
+                                 class_number,
+                                 filters_number,
+                                 network_depth=6,
+                                 use_residual=True,
```

Comment:
These to options (use_residual and use_bottleneck) are not avaialable at the top-level generate_models function (or put in generate_InceptionTime_hyperparameter_set). Maybe that's not necessary, but do we need these options at all?

Reply:
In their paper they tested the influence of both parameters. In most cases they didn't make a lot of difference, but for some they had quite some impact. Still, it seems to minor of a thing to add it to the top-level generate_models_function to me. I would like to leave them in for now (and do some more testing soonish?).